coreapi/main_api.py

```diff
@@ -33,6 +35,77 @@
 logger = RekogntionLogger(name="main_api")
 
 
+def text_detect(image):
+    """     Scene Text Detection
+    Args:
+            *   image: path of image
+    Workflow:
+            *   A numpy array of an image with text is taken as input
+                inference input dimension requires dimension to be in a
+                multiple of 32 therefore the input is first resized to
+                 required input dimension and then normalized.
+            *   Now the processed output is further processed to make it a
+                json format which is compatible to TensorFlow Serving input.
+            *   Then a http post request is made at localhost:8501.
+                The post request contain data and headers.
+            *   Incase of any exception, it return relevant error message.
+            *   output from TensorFlow Serving is further processed using
+                Locality-Aware Non-Maximum Suppression (LANMS)
+            *   A dictionary is created with boxes as the key an coordinates
+                of bounding boxes of text as value
+    Returns:
+            *   Dictionary having boxes as the key and coordinates of bounding
+                boxes of text as value.
+    """
+
+    logger.info(msg="text_detect called")
+    img = cv2.imread(image)[:, :, ::-1]
+    img_resized, (ratio_h, ratio_w) = preprocess(img)
+    img_resized = (img_resized / 127.5) - 1
+    data = json.dumps({"signature_name": "serving_default",
```

Comment:
opencv reads in BGR and you're giving input in BGR instead RGB (assuming data was trained on RGB ) verify it  and also at other places as well.

Reply:
It has been trained on RGB only.
Yes, opencv reads in BGR. That's why "[:, :, ::-1]" has been added to convert to RGB.
Resolving this I guess.

Reply:
use cv2.cvtColor(screen, cv2.COLOR_BGR2RGB) , else whenever an image with alpha channel will come then in preprocessing instead of alpha channel the Blue channel will be removed. or else validate the no of channels here.

Reply:
@PulkitMishra